torchgeo/trainers/utils.py

```diff
@@ -72,7 +72,9 @@ def load_state_dict(model: Module, state_dict: "OrderedDict[str, Tensor]") -> Mo
     expected_in_channels = state_dict["conv1.weight"].shape[1]
 
     num_classes = cast(nn.Module, model.fc).out_features
-    expected_num_classes = state_dict["fc.weight"].shape[0]
+    expected_num_classes = None
+    if "fc.weight" in state_dict:
```

Comment:
"fc.weight" is not in (at least) the `ResNet50_Weights.SENTINEL2_ALL_MOCO` set of weights